packages/circuit-ui/components/TextArea/TextArea.tsx

```diff
@@ -29,10 +36,20 @@ const textAreaStyles = css`
 /**
  * TextArea component for forms.
  */
-export const TextArea = forwardRef(
-  (props: TextAreaProps, ref: TextAreaProps['ref']) => (
-    <Input {...props} inputStyles={textAreaStyles} as="textarea" ref={ref} />
-  ),
+export const TextArea = forwardRef<HTMLCircuitInputElement, TextAreaProps>(
+  (props, passedRef) => {
+    const localRef = useRef<HTMLCircuitInputElement>(null);
+    const modifiedProps = useAutoExpand(localRef, props);
```

Comment:
We don't know if the component is called with a ref,
So we need to make a local ref which we are sure is defined.

And because it's a hook, we can't conditionnally use `useRef` only if `passedRef` is undefined.
It's a pattern I use often, I don't know better way but I can look for it.